### ISSUE TITLE
Wrong Capacity displayed?

### ISSUE THREAD

**Issue by 0lm** [kolinger/rd-usb#40]
hi

i apologize in advance, because this question is a little bit off-topic, because I am not really sure what causes this.

So, I used an UM34C (via bluetooth) together with rd-usb to charge a phone. After it fully charged, I checked the numbers and they said something around 1300mAh. That didn't feel right, because the usage-time of that phone was still good and I am using it daily. The original capacity is labled as 3000mAh.

I repeated the test, but this time, I turned off QuickCharge in the phone's settings. And this time, it said 2400mAh, which sounds more believeable, based my usage experience.

Now, I am not sure what caused this. Is rd-usb buggy sometimes with outputs? Or is even my UM34C defect and not properly working when QuickCharge is enabled?

**Comment by kolinger** [comment]
Hi,
what you describe makes complete sense if specific conditions are met.

For example quick charge is asking charger for higher voltage and thus charger will deliver same or more energy with lower current, because voltage is higher. In this condition you will see same or higher watts delivered, identical Wh in the end but much lower mAh.

Comparing mAh is problematic due to voltage. That's why mWh is better unit to compare capacity since mWh accounts for voltage but mAh doesn't.

Phones generally have single cell li-ion battery with nominal voltage around 3.6-3.85V. Specified capacity in mAh is tied to this nominal voltage. Classic USB charger is using 5V and fast charging can be more like 9V or even more. 

Thus it's impossible to compare mAh between 3.8V, 5V or 9V, mAh _capacity will drop_ as voltage increases yet the **actual mWh capacity is constant**. This is because mAh doesn't represent anything without the voltage.

That's why you will always see much lower mAh value than what manufacturer specifies if you use fast charging and even with standard charging mAh figure will always be lower due to different voltage since 5V is higher than voltage of battery.

For example:
3.85V nominal battery @ 3000mAh has 11550mWh energy
5V USB charger @ 2500mAh delivers 12500mWh of energy
9V fast charger @ 1300mAh delivers 11700mWh of energy.

I'm not sure if your fast charging is using 9V, just guessing but if this is the case then your measurement seems accurate as far as it can be and there is nothing wrong with your meter or phone.

Just multiply nominal voltage with mAh and you get true capacity of battery or true transferred energy. Energy what charger gives out should be always higher since charging is not 100% effective thus part of energy is lost in heat. This doesn't mean your battery has higher capacity than specified but more likely this extra energy is lost in transfer.

There is another catch - phone will not let you discharge battery to real 0%, it may show 0% but battery doesn't have truly 0% since if that was the case the phone couldn't show you anything also discharging down to 0% would reduce battery life that's why if phone says 0% it isn't 0% but more.

That's why your external measurement can't cover whole capacity of the battery since the battery isn't completely discharged to zero and thus you will not see complete 100% charge.

You can calculate effective mAh backwards if you get 2400mAh and you know that charger voltage was 5V, then you calculate difference between 3.85V (nominal voltage of the battery) and then multiple measured mAh capacity by this difference to get effective mAh from point of view of the battery.

For example 2400mAh * (5V / 3.85V) = ~3100mAh

In the end measuring capacity of internal battery is always a estimate and not very accurate due to charging/transfer loses and the fact you don't know the initial state of charge of the battery. So if you calculate 3100mAh and you battery is 3000mAh this makes sense and battery is likely 3000mAh and the rest is lost in heat.

TL;DR
Compare capacity in Wh/mWh instead, not mAh. You can also multiple mAh by difference of voltage between nominal voltage of the battery, standard charging and fast charging and then you get fairly matching mAh figure.

**Comment by kolinger** [comment]
Yes, if I got these results I would assume the story from refurbisher is real.

There is possibility that the battery could have for example 95% of capacity and this method can't reliably prove such discrepancy so you can't see if battery is new or nearly new but you can differentiate old battery (like years old) from new battery. In such case I wouldn't care because if I'm unable to measure the difference then for sure I wouldn't be able to notice the difference in day-to-day usage.

Happy to help.

BTW: if you want longer battery life then don't use fast charging if you don't need to, batteries like slow charging better
